app/controllers/ViewConversations.scala

```diff
@@ -79,33 +96,49 @@ class ViewConversations @Inject()(
     }
   }
 
-  def result(client: String, conversationId: String) = Action.async { implicit request =>
+  def result(client: String, conversationId: String) = Action.async { implicit request: MessagesRequest[AnyContent] =>
     secureMessageFrontendConnector
       .resultPartial(client, conversationId)
-      .map { response =>
-        (response.status, response.body) match {
-          case (OK, body)     => Ok(viewConversationMessages(HtmlFormat.raw(body)))
-          case (NOT_FOUND, _) => NotFound
-          case (_, _)         => ServiceUnavailable
-        }
+      .map {
+        case HtmlPartial.Success(_, content) => Ok(viewConversationMessages(content))
+        case HtmlPartial.Failure(Some(NOT_FOUND), body) =>
+          logger.error(s"[ViewConversations][result] - status code:NOT_FOUND, body:$body")
+          NotFound(body)
+        case HtmlPartial.Failure(Some(status), body) =>
+          logger.error(s"[ViewConversations][result] - status code:$status, body:$body")
+          ServiceUnavailable(body)
+        case HtmlPartial.Failure(_, body) =>
+          logger.error(s"[ViewConversations][result] - body:$body")
+          ServiceUnavailable(body)
       }
       .recover {
-        case _ => InternalServerError
+        case _ => {
+          logger.error(s"[ViewConversations][result] - InternalServerError")
+          InternalServerError
+        }
       }
   }
 
-  def viewLetterOrConversation(id: String) = Action.async { implicit request =>
+  def viewLetterOrConversation(id: String) = Action.async { implicit request: MessagesRequest[AnyContent] =>
     secureMessageFrontendConnector
       .letterOrConversationPartial(id)
-      .map { response =>
-        (response.status, response.body) match {
-          case (200, body) => Ok(viewConversationMessages(HtmlFormat.raw(body)))
-          case (404, _)    => NotFound
-          case (_, _)      => ServiceUnavailable
-        }
+      .map {
+        case HtmlPartial.Success(_, content) => Ok(viewConversationMessages(content))
+        case HtmlPartial.Failure(Some(NOT_FOUND), body) =>
+          logger.error(s"[ViewConversations][viewLetterOrConversation] - status code:NOT_FOUND, body:$body")
+          NotFound(body)
+        case HtmlPartial.Failure(Some(status), body) =>
+          logger.error(s"[ViewConversations][viewLetterOrConversation] - status code:$status, body:$body")
+          ServiceUnavailable(body)
+        case HtmlPartial.Failure(_, body) =>
+          logger.error(s"[ViewConversations][viewLetterOrConversation] - body:$body")
+          ServiceUnavailable(body)
       }
       .recover {
-        case _ => InternalServerError
+        case _ => {
```

Comment:
I wouldn't suppress the error. `case NonFatal(err) => logger.error("bla bla", err)`, so we can see what's going wrong.

Reply:
done